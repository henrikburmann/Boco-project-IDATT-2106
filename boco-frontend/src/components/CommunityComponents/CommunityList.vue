<template>
  <!-- A list conatining all the communities -->
  <p
    v-if="!communities.length"
    class="flex place-content-center text-gray-400 mt-8"
  >
    Ingen grupper
  </p>
  <ul v-else>
    <li v-for="community in communities" :key="community">
      <CommunityListItem
        :community="community"
        :member="member"
        class="border-black"
      />
    </li>
  </ul>
</template>

<script>
import CommunityListItem from "@/components/CommunityComponents/CommunityListItem.vue";

export default {
  name: "CommunityList",
  props: {
    communities: Array,
    member: Boolean,
  },
  components: {
    CommunityListItem,
  },
};
</script>
