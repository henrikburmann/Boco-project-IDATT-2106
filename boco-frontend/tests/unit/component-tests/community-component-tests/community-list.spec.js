import { mount } from "@vue/test-utils";
import CommunityList from "@/components/CommunityComponents/CommunityList.vue";

describe("CommunityList component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(CommunityList, {
      props: {
        communities: [
          {
            communityId: 0,
            name: "string",
            description: "string",
            visibility: 0,
            location: "string",
            picture: "string",
          },
          {
            communityId: 0,
            name: "string",
            description: "string",
            visibility: 0,
            location: "string",
            picture: "string",
          },
        ],
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
