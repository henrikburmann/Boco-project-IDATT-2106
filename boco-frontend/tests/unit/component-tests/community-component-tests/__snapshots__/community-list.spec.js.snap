// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityList component renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- A list conatining all the communities -->
  <ul>
    
    <li>
      <div
        class="border-black"
      >
        
        <!-- Main modal -->
        <!--v-if-->
        
        <div
          class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-2"
        >
          <div
            class="h-3 w-14 flex flex-col justify-center items-center ml-2 mt-4 mb-4 mr-2"
          >
            <img
              alt="Bilde"
              class="rounded-md h-14 w-14 object-contain"
              src="string"
            />
          </div>
          <div
            class="flex-1 pl-1 overflow-hidden"
          >
            <div
              class="font-medium text-gray-800 dark:text-white truncate"
            >
              string
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <svg
              aria-hidden="true"
              class="max-h-6 max-w-6 shrink m-2"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 15v2m-6 4h12a2 2 0 002-2v-6a2 2 0 00-2-2H6a2 2 0 00-2 2v6a2 2 0 002 2zm10-10V7a4 4 0 00-8 0v4h8z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="border-black"
      >
        
        <!-- Main modal -->
        <!--v-if-->
        
        <div
          class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-2"
        >
          <div
            class="h-3 w-14 flex flex-col justify-center items-center ml-2 mt-4 mb-4 mr-2"
          >
            <img
              alt="Bilde"
              class="rounded-md h-14 w-14 object-contain"
              src="string"
            />
          </div>
          <div
            class="flex-1 pl-1 overflow-hidden"
          >
            <div
              class="font-medium text-gray-800 dark:text-white truncate"
            >
              string
            </div>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          >
            <svg
              aria-hidden="true"
              class="max-h-6 max-w-6 shrink m-2"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 15v2m-6 4h12a2 2 0 002-2v-6a2 2 0 00-2-2H6a2 2 0 00-2 2v6a2 2 0 002 2zm10-10V7a4 4 0 00-8 0v4h8z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
    </li>
    
  </ul>
  
</div>
`;
